code/src/pages/MovieDetail.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from 'react'
+
+import { useParams } from 'react-router-dom'
+import { Link } from 'react-router-dom'
+import { BackButton } from 'BackButton'
+
+export const MovieDetail = () => {
+  const { movieId } = useParams();
+  const [currentMovie, setcurrentMovie] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [castList, setCastList] = useState([]);
+
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/movie/${movieId}?api_key=303d50a32fc0419fb55796d006e5d6c2&language=en-US`)
+    .then(res => {
+      if (res.ok) {
+        return res.json();
+      } else {
+        throw new Error('404');
+      }
+    })
+    .then((json) => {
+      setIsLoading(false);
+      setcurrentMovie(json);
+    })
+    .catch(() => {
+      window.location.assign('/404');
+    });
+  },[movieId])
+
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/movie/${movieId}/credits?api_key=303d50a32fc0419fb55796d006e5d6c2&language=en-US`)
+    .then(res => {
+      if (res.ok) {
+        return res.json();
+      } else {
+        throw new Error('404');
+      }
+    })
+    .then((json) => {
+      setCastList(json.cast);
+    })
+    .catch(() => {
+     window.location.assign('/404');
+    });
+  },[movieId])
+
+  return ( <>
+  {!isLoading && currentMovie &&
+    <section 
+      className="movie-detail-container" 
+      key={currentMovie.id}
+      style={{  
+        backgroundImage: `url(https://image.tmdb.org/t/p/w1280${currentMovie.backdrop_path})`,
+        backgroundSize: 'cover',
+        backgroundRepeat: 'no-repeat'
+      }}
+    >
+ 
+      <BackButton text="Go back"></BackButton>
+        <div className="movie-details-wrapper">
+          <img className="movie-detail-poster" src={`https://image.tmdb.org/t/p/w342${currentMovie.poster_path}`} alt={currentMovie.title} />
+            <div className="movie-details-text">
+              <h1 className="movie-detail-title">{currentMovie.title}</h1> 
```

Comment:
I think there should only be one h1 tag in a webpage, for example in the header. This is due to accessibility